internal/xpls/version/version.go

```diff
@@ -0,0 +1,140 @@
+// Copyright 2022 Upbound Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package version
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/Masterminds/semver/v3"
+
+	"github.com/crossplane/crossplane-runtime/pkg/logging"
+
+	"github.com/upbound/up/internal/version"
+)
+
+const (
+	// 5 seconds should be more than enough time.
+	clientTimeout = 5 * time.Second
+	cliURL        = "https://cli.upbound.io/stable/current/version"
+
+	errFailedToQueryRemoteFmt = "query to %s failed"
+	errNotSemVer              = "couldn't covert version to semver"
+)
+
+type client interface {
+	Do(*http.Request) (*http.Response, error)
+}
+
+type defaultClient struct {
+	client http.Client
+}
+
+// Informer enables the caller to determine if they can upgrade their current
+// version of up.
+type Informer struct {
+	client client
+	log    logging.Logger
+}
+
+// NewInformer constructs a new Informer.
+func NewInformer(opts ...Option) *Informer {
+	i := &Informer{
+		log:    logging.NewNopLogger(),
+		client: newClient(),
+	}
+
+	for _, o := range opts {
+		o(i)
+	}
+
+	return i
+}
+
+// Option modifies the Informer.
+type Option func(*Informer)
+
+// WithLogger overrides the default logger for the Informer.
+func WithLogger(l logging.Logger) Option {
+	return func(i *Informer) {
+		i.log = l
+	}
+}
+
+// CanUpgrade queries locally for the version of up, uses the Informer's client
+// to check what the currently published version of up is and returns the local
+// and remote versions and whether or not we could upgrade up.
+func (i *Informer) CanUpgrade() (string, string, bool) {
+	local := version.GetVersion()
+	remote, err := i.getCurrent()
+	if err != nil {
+		i.log.Debug(fmt.Sprintf(errFailedToQueryRemoteFmt, cliURL), "error", err)
+		return "", "", false
+	}
+
+	return local, remote, i.newAvailable(local, remote)
+}
+
+func (i *Informer) newAvailable(local, remote string) bool {
+	lv, err := semver.NewVersion(local)
+	if err != nil {
+		// invalid remote version detected
```

Comment:
```suggestion
		// invalid local version detected
```
should this be in error message?